Dashboard/app/js/lib/components/devices/DevicesTab.jsx

```diff
@@ -0,0 +1,387 @@
+        });
+      }, 500);
+    }
+  }
+
+  setCurrentTable = tableName => {
+    this.setState({ currentTable: tableName });
+    this.setState({ selectedEditGroupId: null });
+    this.setState({ newName: null });
+  };
+
+  // DEVICES LIST
+  selectDevice = (id, deviceIds) => {
+    if (deviceIds.some(deviceId => id === deviceId)) {
+      const filterDevice = deviceIds.filter(deviceId => id !== deviceId);
+      return this.setState({ selectedDeviceIds: [...filterDevice] });
+    }
+    return this.setState({ selectedDeviceIds: [...deviceIds, id] });
+  };
+
+  // Get the property of a selected group
+  dialogGroupSelectionChange = e => {
+    const { code, keyId } = e.target.value.length !== 0 && JSON.parse(e.target.value);
+    this.setState({ dialogGroupSelection: { code, keyId } });
+  };
+
+  addDeviceToGroup = dev => {
+    let devices = [];
+
+    // Find all devices that have the same keyId as in the selectedDeviceIds
+    for (let i = 0; i < dev.length; i++) {
+      const filterDevices = this.state.devices.find(device => device.keyId === dev[i]);
+      devices = [...devices, filterDevices];
+      const devicesInGroup = FLOW.store.filter(FLOW.Device, item => item.get('keyId') === dev[i]);
+
+      devicesInGroup.forEach(item => {
+        item.set('deviceGroupName', this.state.dialogGroupSelection.code);
+        item.set('deviceGroup', this.state.dialogGroupSelection.keyId);
+      });
+    }
+
+    // Adding group property to the selected devices
+    devices.map(item => {
+      item.deviceGroupName = this.state.dialogGroupSelection.code;
+      item.deviceGroup = this.state.dialogGroupSelection.keyId;
+      return item;
+    });
+
+    FLOW.store.commit();
+
+    this.cancelAddToGroup();
+  };
+
+  showAddToGroupDialog = () => {
+    this.setState({ showAddToGroupDialogBool: true });
+  };
+
+  cancelAddToGroup = () => {
+    const select = document.querySelector('#select-group');
+    select.value = '';
+    this.setState({ dialogGroupSelection: null });
+    this.setState({ showAddToGroupDialogBool: false });
+  };
+
+  showRemoveFromGroupDialog = () => {
+    this.setState({ showRemoveFromGroupDialogBool: true });
+  };
+
+  cancelRemoveFromGroup = () => {
+    this.setState({ showRemoveFromGroupDialogBool: false });
+  };
+
+  doRemoveFromGroup = dev => {
+    let devices = [];
+
+    // Find all devices that have the same keyId as in the selectedDeviceIds
+    for (let i = 0; i < dev.length; i++) {
+      const filterDevices = this.state.devices.find(device => device.keyId === dev[i]);
+      devices = [...devices, filterDevices];
+      const devicesInGroup = FLOW.store.filter(FLOW.Device, item => item.get('keyId') == dev[i]);
+
+      devicesInGroup.forEach(item => {
+        item.set('deviceGroupName', null);
+        item.set('deviceGroup', null);
+      });
+    }
+
+    devices.forEach(item => {
+      item.deviceGroupName = null;
+      item.deviceGroup = null;
```

Comment:
Instead of `deviceGroup`, replace it with `deviceGroupId` so that its clear its a number.